### PR TITLE
Major upgrade to TF2 (v2.7)

### PR DESCRIPTION
**Pull request by TjarkMiener**
This is a PR draft for the CTLearn v0.6.0 using Keras with the latest TensorFlow v.2.7.

**ToDos**:
- [ ] Handle the data loader with Keras functionality. Previous data loader handler `input_fn()`  should be replaced with a Keras batch generator.
- [ ] Upgrade monoscopic models
- [ ] Upgrade stereoscopic models
- [ ] Enable usage of multiple GPUs via `tf.distribute.MirroredStrategy`
- [ ] Save only the best model checkpoints
- [ ] Calculate the class weights in the init of the DL1DH reader and apply always the class weights for the classification task
- [ ] Upgrade the yaml environment file. Try to install CUDA 11.X in the environment from Nvidia conda. Update dependencies like DL1DH and ctapipe to Python 3.8+.
- [ ] Set the reconstruction tasks from command line. Automatise the selection of labels and their transformations.
- [ ] Plot model architecture

### CODE REVIEW

**Comment by TjarkMiener** [comment]
I've marked this PR as a draft because there are some things missing. @nietootein regarding the installation I was wondering if we can give it a try to rather install CUDA from the Nvidia conda account in the environment than systemwide? It'd be useful for non-root users.

**Comment by TjarkMiener** [comment]
Ready from my side!

**Review by nietootein — APPROVED** [review]
Thanks so much for this massive upgrade, @TjarkMiener!
